agent/consul/leader_metrics.go

```diff
@@ -12,18 +13,25 @@ import (
 	"github.com/hashicorp/consul/logging"
 )
 
+var metricsKeyMeshCAExpiry = map[string][]string{
+	"root-cert":         {"mesh", "active-root-ca", "expiry"},
+	"intermediate-cert": {"mesh", "active-intermediate-ca", "expiry"},
+}
```

Comment:
I think a couple vars might work better than a map here. Since we have to use a string key for a map we don't get the benefits of the compiler checking correctness for us:

```suggestion

var metricsKeyMeshRootCAExpiry = []string{"mesh", "active-root-ca", "expiry"}
var metricsKeyMeshIntermediateCAExpiry = []string{"mesh", "active-intermediate-ca", "expiry"}

```